pkg/sql/logictest/testdata/logic_test/set_schema

```diff
@@ -251,3 +259,148 @@ user testuser
+
+# Ensure setting the type to the same schema is valid.
+statement ok
+ALTER TYPE typ SET SCHEMA public
+
+# Cannot set schema to one that does not exist.
+statement error pq: schema does_not_exist does not exist
+ALTER TYPE typ SET SCHEMA does_not_exist
+
+# Ensure we cannot set schema if there is already a type with the same name
+# in the schema.
+statement ok
+CREATE TYPE s2.typ AS ENUM ()
+
+statement error pq: type typ already exists in schema s2
+ALTER TYPE typ SET SCHEMA s2
+
+# Ensure we cannot set schema to a virtual schema.
+statement error pq: cannot move objects into or out of virtual schemas
+ALTER TYPE typ SET SCHEMA information_schema
+
+# Ensure we cannot set schema to a temporary schema.
+let $temp_schema
+SELECT schema_name FROM [show schemas] WHERE schema_name LIKE '%pg_temp%'
+
+statement error pq: cannot move objects into or out of temporary schemas
+ALTER TYPE typ SET SCHEMA $temp_schema
+
+# Test setting a type from public to s1.
+
+statement ok
+ALTER TYPE typ SET SCHEMA s1
+
+# Ensure that the type is in the designated schema.
+statement ok
+SELECT 'hello'::s1.typ
+
+# Ensure the type does not exist in the public schema.
+statement error pq: type "public.typ" does not exist
+SELECT 'hello'::public.typ
+
+statement error pq: type "typ" does not exist
+SELECT 'hello'::typ
+
+# Testing setting a type to the public schema.
+
+statement ok
+ALTER TYPE s1.typ SET SCHEMA public
+
+# Check that the type is in the designated schema.
+statement ok
+SELECT 'hello'::public.typ
+
+
+# Ensure the type does not exist in the s1 schema.
+statement error pq: type "s1.typ" does not exist
+SELECT 'hello'::s1.typ
+
+# Test moving from one user defined schema to another.
+statement ok
+CREATE TYPE s1.typ2 AS ENUM ('hello')
+
+statement ok
+ALTER TYPE s1.typ2 SET SCHEMA s2
+
+statement ok
+SELECT 'hello'::s2.typ2
+
+# Ensure the type does not exist in the s1 schema.
+statement error pq: type "s1.typ2" does not exist
+SELECT 'hello'::s1.typ2
+
+statement ok
+CREATE TYPE s1.typ3 AS ENUM ('hello')
+
+# Test ALTER TYPE SET SCHEMA inside a transaction.
+statement ok
+BEGIN
+
+statement ok
+ALTER TYPE s1.typ3 SET SCHEMA s2
+
+statement ok
+SELECT 'hello'::s2.typ3
+
+statement ok
+COMMIT
+
+# ALTER TYPE SET SCHEMA changes should reflect after commit.
+
+statement ok
+SELECT 'hello'::s2.typ3
+
+# Ensure the type does not exist in the s1 schema.
+statement error pq: type "s1.typ3" does not exist
+SELECT 'hello'::s1.typ3
+
+statement ok
+CREATE TYPE s1.typ4 AS ENUM ('hello')
+
+# Ensure set schema gets rolled back if the transaction is canceled.
+statement ok
+BEGIN
+
+statement ok
+ALTER TYPE s1.typ4 SET SCHEMA s2
+
+statement ok
+ROLLBACK
+
+# Ensure that the type is in the original schema.
+statement ok
+SELECT 'hello'::s1.typ4
+
+# Ensure that the type is not in the s2 schema.
+statement error pq: type "s2.typ4" does not exist
+SELECT 'hello'::s2.typ4
+
+# Ensure the user has DROP privilege on the table to set the schema.
+statement ok
```

Comment:
how are these permissions tests working i thought that we don't have permissions yet